src/App.tsx

```diff
@@ -1,10 +1,64 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+
 import { UserWarning } from './UserWarning';
+import { getTodos } from './api/todos';
+import { ErrorInfo } from './components/Errorinfo';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { ErrorType } from './types/ErrorType';
+import { FilterType } from './types/FilterType';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 12017;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState<FilterType>(
+    FilterType.All,
+  );
+  const [errorMsg, setErrorMsg] = useState<ErrorType | null>(null);
+
+  const errorFound = (error: ErrorType) => {
```

Comment:
Function names usually start with a verb
```suggestion
  const showError = (error: ErrorType) => {
```